Makefile

```diff
@@ -205,89 +204,82 @@ CHANGELOG.md:
-	package.json lib \
-	$(BUILD_DIRECTORY)/node-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies \
+	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app/node_modules \
 	| $(BUILD_DIRECTORY)
-	mkdir $@
-	cp $(word 1,$^) $@
-	$(CPRF) $(word 2,$^) $@
-	$(CPRF) $(word 3,$^)/* $@
-
-$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH).js: \
-	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app \
-	| $(BUILD_DIRECTORY)
-	./scripts/build/concatenate-javascript.sh -e lib/cli/etcher.js -b $< -o $@ -m
-
-$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH): \
-	$(BUILD_DIRECTORY)/node-$(TARGET_PLATFORM)-$(TARGET_ARCH)-dependencies \
-	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH).js \
-	| $(BUILD_DIRECTORY) $(BUILD_TEMPORARY_DIRECTORY)
-	mkdir $@
-	./scripts/build/node-package-cli.sh -o $@ -l $</node_modules \
-		-n $(APPLICATION_NAME) \
-		-e $(word 2,$^) \
-		-r $(TARGET_ARCH) \
-		-s $(TARGET_PLATFORM)
+	cp -rf lib $@
+	cp -rf package.json $@
 
+ETCHER_CLI_BINARY = $(APPLICATION_NAME_LOWERCASE)
 ifeq ($(TARGET_PLATFORM),win32)
-	./scripts/build/electron-brand-exe.sh \
-		-f $@/etcher.exe \
-		-n $(APPLICATION_NAME) \
-		-d "$(APPLICATION_DESCRIPTION)" \
-		-v "$(APPLICATION_VERSION)" \
-		-c "$(APPLICATION_COPYRIGHT)" \
-		-m "$(COMPANY_NAME)" \
-		-i assets/icon.ico \
-		-w $(BUILD_TEMPORARY_DIRECTORY)
+ETCHER_CLI_BINARY = $(APPLICATION_NAME_LOWERCASE).exe
 endif
 
-ifeq ($(TARGET_PLATFORM),darwin)
-ifdef CSC_NAME
-	./scripts/build/electron-sign-file-darwin.sh -f $@/etcher -i "$(CSC_NAME)"
-endif
-endif
+$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH): \
+	$(BUILD_DIRECTORY)/$(APPLICATION_NAME)-cli-$(APPLICATION_VERSION)-$(TARGET_PLATFORM)-$(TARGET_ARCH)-app \
+	| $(BUILD_DIRECTORY)
+	mkdir $@
+	$(NPX) pkg --output $@/$(ETCHER_CLI_BINARY) -t node6-$(TARGET_PLATFORM_PKG)-$(TARGET_ARCH) $</$(ENTRY_POINT_CLI)
+	./scripts/build/dependencies-npm-extract-addons.sh \
+		-d $</node_modules \
+		-o $@/node_modules
+# pkg currently has a bug where darwin executables
+# can't be code-signed
+# See https://github.com/zeit/pkg/issues/128
+# ifeq ($(TARGET_PLATFORM),darwin)
+# ifdef CSC_NAME
+#		./scripts/build/electron-sign-file-darwin.sh -f $@/$(ETCHER_CLI_BINARY) -i "$(CSC_NAME)"
+# endif
+# endif
+
+# pkg currently has a bug where Windows executables
+# can't be branded
+# See https://github.com/zeit/pkg/issues/149
+# ifeq ($(TARGET_PLATFORM),win32)
+#		./scripts/build/electron-brand-exe.sh \
+#			-f $@/$(ETCHER_CLI_BINARY) \
+#			-n $(APPLICATION_NAME) \
+#			-d "$(APPLICATION_DESCRIPTION)" \
+#			-v "$(APPLICATION_VERSION)" \
+#			-c "$(APPLICATION_COPYRIGHT)" \
+#			-m "$(COMPANY_NAME)" \
+#			-i assets/icon.ico \
+#			-w $(BUILD_DIRECTORY)
```

Comment:
(when this works again) does it make sense to just download `rcedit.exe` into `$(BUILD_DIRECTORY)`, given that it's not a target that we're building?